sqlmesh/dbt/target.py

```diff
@@ -563,11 +566,108 @@ def to_sqlmesh(self) -> ConnectionConfig:
         )
 
 
+class MSSQLConfig(TargetConfig):
+    """
+    Project connection and operational configuration for the SQL Server (MSSQL) target
+
+    Args:
+        host: The MSSQL server host to connect to
+        port: The MSSQL server port to connect to
+        user: User name for authentication
+        password: User password for authentication
+        login_timeout: The number of seconds to wait for a login to complete
+        query_timeout: The number of seconds to wait for a query to complete
+    """
+
+    type: Literal["sqlserver"] = "sqlserver"
+    host: t.Optional[str] = None
+    server: t.Optional[str] = None
+    port: int = 1433
+    database: str = Field(default="master")
+    schema_: str = Field(default="dbo", alias="schema")
+    user: t.Optional[str]
+    username: t.Optional[str]
+    UID: t.Optional[str]
+    password: t.Optional[str]
+    PWD: t.Optional[str]
+    threads: int = 4
+    login_timeout: t.Optional[int] = None
+    query_timeout: t.Optional[int] = None
+    authentication: t.Optional[str] = "sql"
+    schema_authorization: t.Optional[str] = None  # Not supported by SQLMesh
+
+    # Unused ODBC parameters (SQLMesh uses pymssql instead of ODBC)
+    driver: t.Optional[str] = None
+    encrypt: t.Optional[bool] = None
+    trust_cert: t.Optional[bool] = None
+    retries: t.Optional[int] = None
+
+    # Unused authentication parameters (not supported by pymssql)
+    windows_login: t.Optional[bool] = None  # pymssql doesn't require this flag for Windows Auth
+    tenant_id: t.Optional[str] = None  # Azure Active Directory auth
+    client_id: t.Optional[str] = None  # Azure Active Directory auth
+    client_secret: t.Optional[str] = None  # Azure Active Directory auth
```

Comment:
Is it possible to define these and use the `sql` auth method?

Reply:
You could put them in the profile, but they would be ignored for `sql` auth method.

Or are you asking if dbt-sqlserver errors if they're included when auth is `sql`?

Reply:
No that answers it. Making sure it it was not possible for someone to define these for `sql` and then have them be silently ignored by the adapter. Or at least if we were intended to silently ignore that it was an intention decision.